### PR TITLE
Refractoring code to SOLID

### PR DESCRIPTION
**Pull request by SnakyMz**
Things we have done:

- [ ] Refactor main to Single-responsibility Principle
- [ ] Refactor app to SOLID

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +118,18 @@ You can deploy this project using GitPages:
...
+
+- GitHub: [@githubhandle](https://github.com/IsmaelMastronardi)
+- Twitter: [@twitterhandle](https://twitter.com/IsmaMastronardi)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/ismael-mastronardi-361873271/)
+
+👤 **Mahram Hossain🔭**
+
+- GitHub: [@Mahram Hossain](https://github.com/mhnazary)
+- Twitter: [@Mahram Hossain](https://twitter.com/mh_nazary)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/mh-nazary-515686204/)
```
Comment (Efiamotu-1) [inline README.md]: - [ ] This link is currently broken please kindly provide the appropraite link as it is unprofessional to leave broken links in your readme file :+1: :sparkles:

File: app.rb
Comment (Efiamotu-1) [inline app.rb:1]: - [ ] In your app.rb file to follow the single responsibility rule for best practices  all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. see the example code snippet below
- code snippet
```ruby
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    person_type = gets.chomp
```
- [ ] Please kindly update your code base to follow this rule by implementing the above requirement everywhere in your app where you are requesting for user inputs :+1: :sparkles:

Comment (Efiamotu-1) [inline app.rb:1]: - [ ]  this below line of code in your create_student method should've been placed in the `create_person()` method
```ruby
    @people << Student.new(classroom, age, name, parent_permission: parent_permission)
```

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @team, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @SnakyMz                    👋

Your project is complete! ✔️✔  🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors


- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**